src/main/java/seedu/academyassist/model/person/YearGroup.java

```diff
@@ -31,7 +31,7 @@ public YearGroup(String yearGroup) {
      * Returns true if a given string is a valid year group.
      */
     public static boolean isValidYearGroup(String test) {
-        return test.matches(VALIDATION_REGEX);
+        return test != null && test.matches(VALIDATION_REGEX);
     }
```

Comment:
Good that you notice the validation checking for null exception! And clean coding pattern also!